### PR TITLE
feat(coding standards): add a .editorconfig

### PR DESCRIPTION
**Pull request by leagris**
Basics of a coding standard with a .editorconfig file.

Helps contributors to use consistent text encoding, newline and indentation.

See: https://editorconfig.org/

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+insert_final_newline = true
+tab_size = 4
+trim_trailing_whitespace = true
+
+[*.{bat,ini}]
+end_of_line = crlf
+
+[*.{dtd,gradle,json,mcmeta,md,sh,svg,xml,xsd,xsl,yaml,yml}]
+indent_size = 2
+tab_size = 2
```
Comment (Glease) [inline .editorconfig:21]: 2 space doesn't really align with our current style tbh.

Comment (leagris) [reply .editorconfig]: 2-space indent on these files:
https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/src/main/pack.mcmeta
https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/src/main/resources/assets/ic2/sounds.json
https://github.com/GTNewHorizons/GT5-Unofficial/blob/master/.github/workflows/test-scala-presence.yml

Ill remove the .gradle which uses 4-space indent
The other text files ca contain deeply nested elements \(so they are better not using too much spaces on indentation\) or are interpreting indentation as 2-space like markdown lists.

**Review by Glease — APPROVED** [review]
It's not like we will use xml any time soon, but this is ok to me.

You might want this to be in https://github.com/GTNewHorizons/ExampleMod1.7.10 as well.

**Comment by leagris** [comment]
> It's not like we will use xml any time soon, but this is ok to me.
```sh
find . -type f -name '*.xml'
./build/tmp/recompSrc/log4j2.xml
./.idea/vcs.xml
./.idea/git_toolbox_prj.xml
./.idea/inspectionProfiles/Project_Default.xml
./.idea/gradle.xml
./.idea/uiDesigner.xml
./.idea/misc.xml
./.idea/scala_compiler.xml
./.idea/compiler.xml
./.idea/libraries-with-intellij-classes.xml
./.idea/jarRepositories.xml
./.idea/codeStyles/codeStyleConfig.xml
./.idea/codeStyles/Project.xml
./.idea/workspace.xml
```
> 
> You might want this to be in https://github.com/GTNewHorizons/ExampleMod1.7.10 as well.

Absolutely and in all GTNH repositories

**Comment by boubou19** [comment]
all the xml shown look like they are auto genned by idea tbh. Shouldn't we yeet them or is there a real need for those? because i don't think eclipse users will ever use those

**Comment by leagris** [comment]
> all the xml shown look like they are auto genned by idea tbh. Shouldn't we yeet them or is there a real need for those? because i don't think eclipse users will ever use those

These are auto-generated and ignored by the VCS config. Although other projects may have xml or other text file types in there VCS tree, or we may have xml or other text file in the future, so a text editor configuration is appropriate to have anyway.

**Pull request merged**